File: lib/java/opentoken/src/test/java/com/truveta/opentoken/tokentransformer/EncryptionConstantsTest.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.tokentransformer;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import org.junit.jupiter.api.Test;

class EncryptionConstantsTest {

    @Test
    void testEncryptionConstantsExist() {
        assertNotNull(EncryptionConstants.AES);
        assertNotNull(EncryptionConstants.ENCRYPTION_ALGORITHM);
        assertNotNull(EncryptionConstants.IV_SIZE);
    }

    @Test
    void testEncryptionConstantsValues() {
        assertEquals("AES", EncryptionConstants.AES);
        assertEquals("AES/GCM/NoPadding", EncryptionConstants.ENCRYPTION_ALGORITHM);
        assertEquals(32, EncryptionConstants.KEY_BYTE_LENGTH);
        assertEquals(12, EncryptionConstants.IV_SIZE);
        assertEquals(128, EncryptionConstants.TAG_LENGTH_BITS);
    }
}

File: lib/python/opentoken/src/test/opentoken/tokentransformer/encryption_constants_test.py
"""
Copyright (c) Truveta. All rights reserved.
"""
from opentoken.tokentransformer.encryption_constants import EncryptionConstants


class TestEncryptionConstants:
    """Test cases for EncryptionConstants."""

    def test_encryption_constants_exist(self):
        """Test that all encryption constants exist."""
        assert hasattr(EncryptionConstants, 'AES')
        assert hasattr(EncryptionConstants, 'ENCRYPTION_ALGORITHM')
        assert hasattr(EncryptionConstants, 'KEY_BYTE_LENGTH')
        assert hasattr(EncryptionConstants, 'IV_SIZE')
        assert hasattr(EncryptionConstants, 'TAG_LENGTH_BITS')
        assert hasattr(EncryptionConstants, 'TAG_LENGTH_BYTES')

    def test_encryption_constants_values(self):
        """Test that encryption constants have correct values."""
        assert EncryptionConstants.AES == "AES"
        assert EncryptionConstants.ENCRYPTION_ALGORITHM == "AES/GCM/NoPadding"
        assert EncryptionConstants.KEY_BYTE_LENGTH == 32
        assert EncryptionConstants.IV_SIZE == 12
        assert EncryptionConstants.TAG_LENGTH_BITS == 128
        assert EncryptionConstants.TAG_LENGTH_BYTES == 16

    def test_encryption_constants_immutable(self):
        """Test that encryption constants are not None."""
        assert EncryptionConstants.AES is not None
        assert EncryptionConstants.ENCRYPTION_ALGORITHM is not None
        assert EncryptionConstants.KEY_BYTE_LENGTH is not None
        assert EncryptionConstants.IV_SIZE is not None
        assert EncryptionConstants.TAG_LENGTH_BITS is not None
        assert EncryptionConstants.TAG_LENGTH_BYTES is not None
